uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -410,5 +410,131 @@ define([
         });
       });
     });
+
+    describe('check status methods', function () {
+
+      var optionGroupDeferred,
+        mockOptionValue,
+        promise;
+
+      function commonSetup(statusName, methodName) {
+        optionGroupDeferred = $q.defer();
+        mockOptionValue = [{
+          name: 'approved',
+          value: '1'
+        }, {
+          name: 'cancelled',
+          value: '2'
+        }, {
+          name: 'rejected',
+          value: '3'
+        }, {
+          name: 'waiting_approval',
+          value: '4'
+        }, {
+          name: 'more_information_requested',
+          value: '5'
+        }];
+
+        spyOn(OptionGroup, 'valuesOf').and.returnValue(optionGroupDeferred.promise);
+
+        optionGroupDeferred.resolve(mockOptionValue);
+
+        LeaveRequestInstance.status_id = mockOptionValue.find(function (option) {
+          return option.name === statusName
+        }).value;
+
+        promise = LeaveRequestInstance[methodName]();
+      }
+
+      afterEach(function () {
+        $rootScope.$apply();
+      });
+
+      describe('isApproved', function () {
+
+        it('return true if status is approved', function () {
+          commonSetup('approved', 'isApproved');
```

Comment:
I think this whole setup does not "read" as well as it could, everything is encapsulated in the `commonSetup` function, which has both a name and a signature that is not really clear. To me the following is better, it might be a bit more verbose but it's also clearer and thus a better documentation
```js
describe('check status methods', function () {
  var promise;

  beforeEach(function () {
    spyOn(OptionGroup, 'valuesOf').and.callFake(function () {
      return $q.resolve(optionGroupMock.getCollection('hrleaveandabsences_leave_request_status'));
    });
  });

  describe('isApproved()', function () {
    describe('when the leave request is approved', function () {
      beforeEach(function () {
        LeaveRequestInstance.status_id = valueOfRequestStatus('approved');
        promise = LeaveRequestInstance.isApproved();
      });
      
      it('returns true', function () {
        promise.then(function (result) {
          expect(result).toBe(true);
        });
      });
    });

    describe('when the leave request is not approved', function () {
      beforeEach(function () {
        LeaveRequestInstance.status_id = valueOfRequestStatus('cancelled');
        promise = LeaveRequestInstance.isApproved();
      });

      it('returns false', function () {
        promise.then(function (result) {
          expect(result).toBe(false);
        });
      });
    });
  });

  // ...
  
  function valueOfRequestStatus(statusName) {
    // ...
  }
});
```